### DISCUSSION TITLE
New features?

### DISCUSSION THREAD

**Discussion by AustenLamacraft** [tripos-education/maths-tripos-questions#30] [Ideas]
What new features would you like to see? Specific examples of other places where these features are available are helpful. 

A couple of ideas to get the ball rolling:

<details>
  <summary>1. Spoilers</summary>
  
This is already available in GitHub Flavored Markdown. Unfortunately, having to render the LaTeX in comments in the browser means it isn't available on the site. Apparently, [LaTeX is coming (back) to GitHub](https://github.com/github/feedback/discussions/2816#discussioncomment-2664420), so this problem may solve itself if we can drop the custom render.
  
</details>

2. A social media card for each question, which can display some part of the question so people can get an idea of whether they are interested if the link is posted on Twitter / Discord / whatever.

**Comment by thenewtonium** [comment]
Wrt spoilers: any idea if the [@traptitech/markdown-it-spoiler](https://www.npmjs.com/package/@traptitech/markdown-it-spoiler) would work?

**Reply by AustenLamacraft** [reply]
I think it would be best to keep the GH syntax. Changing from one to another would lead to breaking spoilers. I’ll look into whether the GH syntax can be made to work.

**Reply by thenewtonium** [reply]
Hmm fair, though surely allowing LaTeX is already deviating from GH syntax?

**Reply by AustenLamacraft** [reply]
True, but that's essential, and may go away if GH fulfil their promise to render LaTeX. Since spoilers already exist on GH, we should probably follow the syntax (otherwise you'd have the problem of spoilers being visible here on discussion but not on the site, or vice versa)

**Reply by AustenLamacraft** [reply]
Spoilers à la GH should now be working. Give it a go!

**Reply by thenewtonium** [reply]
<details><summary></summary>
Epic

<details><summary>Out of interest...</summary>
...what was causing it to not render properly before?
</details>

</details>

**Reply by AustenLamacraft** [reply]
I'm not actually sure. I swapped out the markup rendering for [micromark](https://github.com/micromark/micromark) which has an allow HTML option, and that seemed to do it.

**Comment by callumhobbis** [comment]
Could you make a small edit to FAQ to include information on how to edit / delete comments by going to the main GitHub page with the comment on it? I've noticed some people have brought it up and don't realise it _is_ possible.

**Reply by AustenLamacraft** [reply]
It does say so but I can make it more explicit. You could also add a comment on the FAQ pointing that out

**Comment by lorcaheeney** [comment]
What about adding the ability to search questions for specific text when filtering them by year and course? I implemented a similar thing here: https://lorcajheeney.github.io/qfind/search.html for alevel questions.

**Reply by AustenLamacraft** [reply]
Thanks for the link! Yes, that makes sense, although I guess you can just visit the course page e.g. https://questions.tripos.org/courses/applications%20of%20quantum%20mechanics/ where all the questions for a course are displayed. 

It looks like it's getting to the point where searching past comments may be needed...

**Comment by thenewtonium** [comment]
Some ideas...
1. Separation of current and discontinued courses (particularly on the part ii page).
2. Some "summary text" (not the right word but I can't think what to call it) on course pages with links to resources etc. (maybe a comments widget on the page so people can suggest resources?) On newer courses it could link to older courses with similar content and vice versa, for example.
